### DISCUSSION TITLE
nextcloud login with zitadel not using username

### DISCUSSION THREAD

**Discussion by hobyte** [zitadel/zitadel#5649] [Q&A] [answered]
I'm currently trying to set up nextcloud with zitadel openid login. So far I followed [this guide](https://zitadel.com/blog/zitadel-as-sso-provider-for-selfhosting) to set up the nextcloud social login app and configure the login with zitadel. It all works, bit I have one problem: In nextcloud, the zitadel userId is used as username. I cannot figure out how I can configure zitadel/nextcloud to use the zitadel username in nextcloud and not the userId.
These are my settings in nextcloud:
![grafik](https://user-images.githubusercontent.com/63231858/230987584-5e79bbc6-b37f-4c5e-8721-d7d6d3e1e4d0.png)

**Comment by fforootd** [comment, marked as answer]
Hm I think if you want to control more aspects of the mapping you need to work with https://github.com/pulsejet/nextcloud-oidc-login
This is actually a fork of the default lib but it allows to configure custom mappings.

In you case you could then map the "preferred_username" from zitadel to nextcloud.

Be aware that "usernames" in general are not how zitadel (and other IdPs) uniquely identify objects.
That is why the "sub" claim is an ID.

More about the claims we have can be found here https://zitadel.com/docs/apis/openidoauth/claims

**Reply by Huber1** [reply]
Shouldn't this also work with Display name claim field?
It doesn't work when i put in `email` but if i understood it correctly it should work somehow

**Reply by fforootd** [reply]
I am not really sure how nextcloud treats this but from our end it should 😁
